src/services/Store/Store.ts

```diff
@@ -0,0 +1,48 @@
+import EventBus from "../EventBus";
+import set from "../../utils/set";
+import IStore from "./store-interfaces/IStore";
+import IUser from "./store-interfaces/IUser";
+import IChat from "./store-interfaces/IChat";
+import IMessage from "./store-interfaces/IMessage";
+import AuthAPIController from "../../utils/controllers/auth-api-controller";
+import ChatsAPIController from "../../utils/controllers/chats-api-controller";
+
+export enum StoreEvents {
+    updated = 'updated'
+};
+
+const authAPIController = new AuthAPIController();
+const chatsAPIController = new ChatsAPIController();
+
+class Store extends EventBus {
+    private state: IStore = {
+        user: {},
+        chats: [],
+        currentChatID: -1,
+        messages: []
+    };
```

Comment:
было бы неплохо сделать Store дженерик-классом, т.е. вот так

```suggestion
class Store<T> extends EventBus {
    private state: T;
    
    init(defaultState: T) {
      this.state = defaultState;
    }
```